From d966eca4a05ac333b6c1300466e87ce9907d35f0 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 19 Apr 2019 12:21:04 +0300
Subject: [PATCH] [SW-0] pll: Gaudi set ALL PLLs

This way we can easily skip PLL init in u-boot.
No need to figure out which PLL is bypassed or not

Change-Id: I8cf4dff275181ecfdfb13eee48939338aac78ba3
---

diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index 646ff84..e609ddb 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -35,15 +35,93 @@
 }
 
 static struct hl_pll_cfg gaudi_pll_cfg[] = {
-	{
-		.name = "CPU", .base = mmPSOC_CPU_PLL_BASE,
-		.nr = 0, .nf = 47, .od = 1, .nb = 23,
-		.div_factor = { 0, 1, 47, -1 }
-	}, {
-		.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
-		.nr = 4, .nf = 331, .od = 3, .nb = 165,
-		.div_factor = { 0, 0, 4, 9 }
-	},
+		{
+			.name = "CPU", .base = mmPSOC_CPU_PLL_BASE,
+	//		.nr = 0, .nf = 47, .od = 1, .nb = 23,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 47, -1 }
+		}, {
+			.name = "DMA_W", .base = mmDMA_W_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "IF_W", .base = mmIF_W_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "IF_E", .base = mmIF_E_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "DMA_E", .base = mmDMA_E_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "MESH_W", .base = mmMESH_W_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "SRAM_W", .base = mmSRAM_W_PLL_BASE,
+	//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, -1, -1, -1 }
+		}, {
+			.name = "MESH_E", .base = mmMESH_E_PLL_BASE,
+	//		.nr = 0, .nf = 59, .od = 1, .nb = 29,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "SRAM_E", .base = mmSRAM_E_PLL_BASE,
+	//		.nr = 0, .nf = 35, .od = 1, .nb = 17,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, -1, -1, -1 }
+		}, {
+			.name = "MME", .base = mmPSOC_MME_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
+	//		.nr = 0, .nf = 37, .od = 1, .nb = 18,
+			.nr = 4, .nf = 331, .od = 3, .nb = 165, //DAY 1 (830MHz)
+			.div_factor = { 0, 0, 4, 9 }
+		}, {
+			.name = "TPC", .base = mmPSOC_TPC_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "HBM", .base = mmPSOC_HBM_PLL_BASE,
+			.nr = 0, .nf = 37, .od = 1, .nb = 18,
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "NIC0", .base = mmNIC0_PLL_BASE,
+			.nr = 24, .nf = 1665, .od = 3, .nb = 832,
+			.div_factor = { 0, 4, 8, -1 }
+		}, {
+			.name = "NIC1", .base = mmNIC1_PLL_BASE,
+			.nr = 24, .nf = 1665, .od = 3, .nb = 832,
+			.div_factor = { 0, 4, 8, -1 }
+		}, {
+			.name = "NIC2_HBM", .base = mmNIC2_HBM_PLL_BASE,
+			.nr = 0, .nf = 37, .od = 1, .nb = 18,
+			.div_factor = { 0, 1, -1, -1 }
+		}, {
+			.name = "NIC2_MME", .base = mmNIC2_MME_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		}, {
+			.name = "NIC2_TPC", .base = mmNIC2_TPC_PLL_BASE,
+	//		.nr = 0, .nf = 63, .od = 1, .nb = 31,
+			.nr = 0, .nf = 31, .od = 15, .nb = 15, //DAY 1 (100 MHz)
+			.div_factor = { 0, 1, 2, 7 }
+		},
 };
 
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
diff --git a/gaudi/gaudi.h b/gaudi/gaudi.h
index 6f80dc9..37564b4 100644
--- a/gaudi/gaudi.h
+++ b/gaudi/gaudi.h
@@ -8,8 +8,24 @@
 /* Use DumpSel[11] to skip PCIe initialization */
 #define SKIP_PCIE_INIT BIT(19)
 
+#define mmDMA_W_PLL_BASE		0x487000
+#define mmIF_W_PLL_BASE			0x488000
+#define mmIF_E_PLL_BASE			0x4A7000
+#define mmDMA_E_PLL_BASE		0x4A8000
+#define mmMESH_W_PLL_BASE		0x4C7000
+#define mmSRAM_W_PLL_BASE		0x4C8000
+#define mmMESH_E_PLL_BASE		0x4E7000
+#define mmSRAM_E_PLL_BASE		0x4E8000
 #define mmPSOC_CPU_PLL_BASE		0xC70000
+#define mmPSOC_MME_PLL_BASE		0xC71000
 #define mmPSOC_PCI_PLL_BASE		0xC72000
+#define mmPSOC_TPC_PLL_BASE		0xC73000
+#define mmPSOC_HBM_PLL_BASE		0xC74000
+#define mmNIC0_PLL_BASE			0xCF9000
+#define mmNIC1_PLL_BASE			0xD39000
+#define mmNIC2_HBM_PLL_BASE		0xD78000
+#define mmNIC2_MME_PLL_BASE		0xD79000
+#define mmNIC2_TPC_PLL_BASE		0xD7A000
 #define CPU_PLL_BASE			mmPSOC_CPU_PLL_BASE
 
 #define HL_CPU_CYCELS_PER_USEC		120000000
